packages/types/src/DidDetails.ts

```diff
@@ -139,75 +145,46 @@ export type DidServiceEndpoint = {
   /**
    * The identifier of the endpoint, without the leading did:kilt:<did_identifier> prefix.
    */
-  id: string
+  id: UriFragment
   /**
    * A list of service types the endpoint exposes.
    */
-  types: string[]
+  type: string[]
   /**
    * A list of URIs the endpoint exposes its services at.
    */
-  uris: string[]
+  serviceEndpoint: string[]
 }
 
 /**
- * An internal representation of data associated with a DID, equivalent to a DID document.
+ * A signature issued with a DID associated key, indicating which key was used to sign.
  */
-export interface IDidDetails {
-  /**
-   * The decentralized identifier (DID) to which the remaining info pertains.
-   */
-  uri: DidUri
+export type DidSignature = {
+  keyUri: DidResourceUri
+  signature: string
+}
 
-  /**
-   * Retrieves a particular public key record via its id.
-   *
-   * @param id The key ID, without the leading DID URI.
-   * @returns [[DidKey]] or undefined if no key with this id is present.
-   */
-  getKey(id: DidKey['id']): DidKey | undefined
+export interface DidDetails {
+  identifier: DidIdentifier
+  uri: DidUri
 
-  /**
-   * Retrieves public key details from the [[IDidDetails]].
-   *
-   * @param relationship A [[KeyRelationship]] or 'none' to filter out keys with a specific key
-   * relationship, undefined to return all keys.
-   * @returns An array of all or selected [[DidVerificationKey]]s, depending on the `relationship` parameter.
-   */
-  getVerificationKeys(
-    relationship: VerificationKeyRelationship
-  ): DidVerificationKey[]
+  authentication: [DidVerificationKey]
+  assertionMethod?: [DidVerificationKey]
+  capabilityDelegation?: [DidVerificationKey]
+  keyAgreement?: DidEncryptionKey[]
 
-  /**
-   * Retrieves public key details from the [[IDidDetails]].
-   *
-   * @param relationship A [[KeyRelationship]] or 'none' to filter out keys with a specific key
-   * relationship, undefined to return all keys.
-   * @returns An array of all or selected [[DidEncryptionKey]]s, depending on the `relationship` parameter.
-   */
-  getEncryptionKeys(relationship: EncryptionKeyRelationship): DidEncryptionKey[]
+  service?: DidServiceEndpoint[]
+}
 
-  getKeys(): DidKey[]
+export interface DidCreationDetails {
```

Comment:
Since this is only used to create a Full did, I think the name should reflect that.

Reply:
I have put on a separate branch my proposal to improve that https://github.com/KILTprotocol/sdk-js/commit/dcb30ee5060f8d9545b545d9d7d7b2d779a5e139

Reply:
Added in https://github.com/KILTprotocol/sdk-js/pull/585/commits/398758776955a682e81c9cc8b46291f96aebbf35